### PR TITLE
Feature design administration space

### PR DESCRIPTION
**Pull request by fastndcurious**

### CODE REVIEW

File: app/controllers/administration_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 class AdministrationController < ApplicationController
+  # binding.pry
   before_action :authenticate_admin!
   def index; end
```
Comment (stephan-dev) [inline app/controllers/administration_controller.rb:6]: je pense qu'on peut virer def index; end

File: app/controllers/items_controller.rb [hunk 2]
```diff
@@ -1,11 +1,12 @@
 # frozen_string_literal: true
 
 class ItemsController < ApplicationController
+  # binding.pry
   def index
     @items = Item.all
   end
 
-  def show;
+  def show
```
Comment (stephan-dev) [inline app/controllers/items_controller.rb]: même modif sur les deux PR (le point virgule)

File: app/models/item.rb [hunk 3]
```diff
@@ -17,8 +17,13 @@
...
 
 class Item < ApplicationRecord
+  # binding.pry
   has_many :item_categories, dependent: :destroy
   has_many :item_wishlists, dependent: :destroy
   has_many :categories, through: :item_categories
   has_many :wishlists, through: :item_wishlists
+
+  def original_price
+    1.00
```
Comment (stephan-dev) [inline app/models/item.rb]: c'est temporaire j'imagine?

**Comment by stephan-dev** [comment]
good for me, as long as the create item methods creates item with all the required params (name, desc...).
Expect small conflicts with latest PR (see comments) (we both removed a ";" , etc)

File: app/models/item.rb (hunk 3 above)
Comment (azuryue) [reply app/models/item.rb]: oui, et l'auto-tabulation m'a crée un `original_price` au lieu d'un `price`

File: app/controllers/administration/items_controller.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Administration
+  class ItemsController < ApplicationController
+    # bindin.pry
```
Comment (Crysicia) [inline app/controllers/administration/items_controller.rb]: Useless comment detected

File: app/controllers/administration/items_controller.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      if item.update
+        flash[:notice] = "Item was successfully updated!"
+        redirect_to items_path
+      else
+        render 'edit'
+      end
+    end
+
+    def destroy
+      @item = @item.find(params[:id])
```
Comment (Crysicia) [inline app/controllers/administration/items_controller.rb]: I think it should be `@item = Item.find(params[:id])` instead.

File: app/controllers/administration_controller.rb [hunk 6]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 class AdministrationController < ApplicationController
+  # binding.pry
```
Comment (Crysicia) [inline app/controllers/administration_controller.rb]: Useless comment here.

File: app/controllers/administration_controller.rb (hunk 1 above)
Comment (Crysicia) [reply app/controllers/administration_controller.rb:6]: No, it tells us that we have an index view.

File: app/controllers/items_controller.rb [hunk 7]
```diff
@@ -1,11 +1,12 @@
 # frozen_string_literal: true
 
 class ItemsController < ApplicationController
+  # binding.pry
```
Comment (Crysicia) [inline app/controllers/items_controller.rb:4]: Useless comment.

File: app/models/item.rb [hunk 8]
```diff
@@ -17,8 +17,13 @@
 #
 
 class Item < ApplicationRecord
+  # binding.pry
```
Comment (Crysicia) [inline app/models/item.rb]: Useless comment.

File: app/models/item.rb (hunk 3 above)
Comment (Crysicia) [reply app/models/item.rb]: Should be fixed.

**Review by Crysicia — CHANGES_REQUESTED** [review]

File: app/controllers/administration/items_controller.rb (hunk 5 above)
Comment (azuryue) [reply app/controllers/administration/items_controller.rb]: lol dumb mistake, thanks !

File: app/controllers/administration_controller.rb (hunk 6 above)
Comment (azuryue) [reply app/controllers/administration_controller.rb]: yes & no, pry is Life. will be removed after

File: app/controllers/administration_controller.rb (hunk 1 above)
Comment (azuryue) [reply app/controllers/administration_controller.rb:6]: yep agree with @Crysicia

File: app/controllers/administration/items_controller.rb [hunk 9]
```diff
@@ -2,7 +2,7 @@
 
 module Administration
   class ItemsController < ApplicationController
-    # bindin.pry
+    # binding.pry
```
Comment (Crysicia) [inline app/controllers/administration/items_controller.rb]: Useless comment

File: app/controllers/administration/items_controller.rb [hunk 10]
```diff
@@ -43,15 +43,15 @@ def update
     end
 
     def destroy
-      @item = @item.find(params[:id])
+      @item = item.find(params[:id])
```
Comment (Crysicia) [inline app/controllers/administration/items_controller.rb]: Should have a capital `I`

File: app/controllers/administration_controller.rb [hunk 11]
```diff
@@ -3,5 +3,13 @@
...
   before_action :authenticate_admin!
+  layout 'administration'
   def index; end
+
+  private
+
+  def authenticate_admin!
+    authenticate_user!
+    redirect_to root_url unless current_user.admin
+  end
```
Comment (Crysicia) [inline app/controllers/administration_controller.rb]: Doesn't need to create a custom method here, devise already handle dynamics helpers

File: app/assets/stylesheets/administration.scss [hunk 12]
```diff
@@ -0,0 +1 @@
+/* integrate Tarek template here */
```
Comment (stephan-dev) [inline app/assets/stylesheets/administration.scss:1]: if we do like this, then we just add " *= require app " in this file

File: app/controllers/administration_controller.rb (hunk 11 above)
Comment (azuryue) [reply app/controllers/administration_controller.rb]: ok

**Review by Crysicia — APPROVED** [review]

**Review by stephan-dev — APPROVED** [review]
approved

**Pull request merged**